tools/cldr-apps/src/test/java/org/unicode/cldr/unittest/web/TestSTFactory.java

```diff
@@ -508,434 +522,501 @@ public void handlePathValue(String path, String value) {
-                                                        uLevel.canManageSomeUsers()
-                                                                && uLevel.isManagerFor(
-                                                                        u.getOrganization(),
-                                                                        onLevel,
-                                                                        onUser.getOrganization());
-                                                assertEquals(
-                                                        "New(ex) vs old(got) manage test: "
-                                                                + uLevel
-                                                                + "/"
-                                                                + onLevel,
-                                                        newTest,
-                                                        oldTest);
-                                                actualResult = actualResult && newTest;
-                                            }
-                                            break;
-                                        default:
-                                            errln("Unhandled action: " + action);
-                                    }
-                                    assertEquals(
-                                            u.org
-                                                    + ":"
-                                                    + uLevel
-                                                    + " "
-                                                    + action
-                                                    + " "
-                                                    + onUser.org
-                                                    + ":"
-                                                    + onLevel,
-                                            allowed,
-                                            actualResult);
-                                }
+                                handleElementVerifyUser(attrs);
                                 break;
                             case "echo":
                             case "warn":
-                                if (value == null) {
-                                    logln("*** " + elem + "  \"" + "null" + "\"");
-                                } else {
-                                    logln("*** " + elem + "  \"" + value.trim() + "\"");
-                                }
+                                handleElementEcho(value, elem);
                                 break;
                             default:
                                 throw new IllegalArgumentException(
                                         "Unknown test element type " + elem);
                         }
                     }
 
+                    private void handleElementVerify(
```

Comment:
this code is too deeply nested! the root of the problem starts with `myReader.setHandler(...)` followed by all the code for the unnamed handler being within the parentheses. it would be sooooo much better to give the handler a name, and just put its name in the parentheses, and define the handler as a named class outside of `runDataDrivenTest` to prevent `runDataDrivenTest` from being 600+ lines long

the deep nesting probably causes the formatter to insert more newlines, adding insult to injury